eth/db/storage.py

```diff
@@ -112,8 +118,12 @@ def __delitem__(self, key: bytes) -> None:
         try:
             del write_trie[hashed_slot]
         except trie_exceptions.MissingTrieNode as exc:
-            self.logger.warning(exc)
-            self.logger.debug("expected to be able to delete slot, but:", exc_info=True)
+            raise MissingStorageTrieNode(
+                exc.missing_node_hash,
+                self._starting_root_hash,
+                exc.requested_key,
+                self._address,
```

Comment:
Note that we don't actually *need* the address pre-image in Beam Sync. But it seemed silly to send the hash here when we have the address in natural form already.